src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -309,19 +309,6 @@ private int RootSlot(int slot)
             }
         }
 
-#if DEBUG
-        protected override void VisitRvalue(BoundExpression node, bool isKnownToBeAnLvalue = false)
```

Comment:
>VisitRvalue [](http://example.com/codeflow?start=32&length=11)

I think we should keep this method in its original form. Initial binding should be changed instead to maintain the invariant.